kobo/django/views/generic.py

```diff
@@ -3,10 +3,54 @@
 import warnings
 from django.conf import settings
 from django.db.models.query import QuerySet
+from django.http import HttpResponse
+from django.template.loader import get_template
 from django.views.generic.edit import ProcessFormView, FormMixin
 from django.views.generic.list import ListView
 from django.views.generic.detail import DetailView
 
+
+class UsersAclMixin:
+    def _check_acl_permission(self, request):
+        permission_type = getattr(settings, "USERS_ACL_PERMISSION", "")
+
+        if permission_type == "authenticated" and not request.user.is_authenticated:
```

Comment:
My intention is to keep everything as it is if not explicitly configured. If by default to use the most secure option, it may potentially break the apps depending on kobo, no? I could change the default behavior if this is not the case.

Reply:
I agree, but I'm not talking about what happens by default if the service doesn't specify a setting.  I'm talking about what happens if they specify an incorrect setting.

The scenario is something like this.  The owner of a kobo-using service puts this in settings:

```
USERS_ACL_PERMISSION = "authetnicated"
```

Notice the word is spelled incorrectly.  What should happen now?

What you've currently implemented is that it behaves the same as if the setting was set to `""`, i.e. the least secure option.  I think this is [CWE-636: Not Failing Securely ('Failing Open')](https://cwe.mitre.org/data/definitions/636.html).

I'm suggesting that it should instead just crash, or if you're worried about that, it should instead behave as though the most secure option was specified.

Reply:
I now get it, thanks for the clarification, addressed in 7dbc953